src/software/thunderscope/replay/test/replay_corruption_test.py

```diff
@@ -0,0 +1,189 @@
+
+    :return: the proto that we are referencing
+    """
+    some_random_proto = RobotId(id=1, team=2)
+    return some_random_proto 
+
+def create_valid_log_entry(proto: Message, current_time: float) -> str:
+    """
+    creating normal log entries
+
+    :proto: the reference proto
+    :current_time: the reference time
+    :return: the log entries that the replay file format uses
+    """
+    serialized_proto = base64.b64encode(proto.SerializeToString())
+    log_entry = (
+        f"{current_time}{REPLAY_METADATA_DELIMETER}"
+        + f"{proto.DESCRIPTOR.full_name}{REPLAY_METADATA_DELIMETER}"
+        + f"{serialized_proto}\n"
+    )
+    return log_entry
+
+def create_missing_delimeter_log_entry(proto: Message, current_time: float) -> str: 
+    """
+    creating corrupted log entries
+
+    :proto: the reference proto
+    :current_time: the reference time
+    :return: the log entries that the replay file format uses
+    """
+    serialized_proto = base64.b64encode(proto.SerializeToString())
+    return (
+        f"{current_time}"
+        + f"{proto.DESCRIPTOR.full_name}{REPLAY_METADATA_DELIMETER}"
+        + f"{serialized_proto}\n"
+    )
+
+def create_corrupt_log_entry(proto: Message, current_time: float) -> str: 
+    """
+    creating corrupted log entries
+
+    :proto: the reference proto
+    :current_time: the reference time
+    :return: the log entries that the replay file format uses
+    """
+    serialized_proto = base64.b64encode(proto.SerializeToString())
+    return (
+        f"{current_time}{REPLAY_METADATA_DELIMETER}"
+        + f"{proto.DESCRIPTOR.full_name}thisissomethingthatwouldbebad{REPLAY_METADATA_DELIMETER}"
+        + f"{serialized_proto}thisisalsosomethingbad\n"
+    )
+
+
+def make_part_replay_chunks(list_of_protos: [Message], save_path:str, duration:float, start_time: float, gen_log_entry_func: Callable[[Message, float], None], frequeny=0.1):
+    """
+    making a part of the replay chunks and appending it to the 0.replay file. There would be a frequency% chance that a
+    invalid log  entry are created
+
+
+    :list_of_protos: the list of proto that we are referencing when creating the log entries
+    :save_path: where we are saving the replay file
+    :duration: how long do we want t create this replay chunks 
+    :start_time: when is the replay chunk being started? 
+    :gen_log_entry_func: the function that is used to generate invalid log entries 
+    :frequency: what percent of the time should we call gen_log_entry_func
+    """
+    os.makedirs(save_path, exist_ok=True)
+    path_to_repaly_file = os.path.join(save_path, "0.replay")
+
+    with gzip.open(path_to_repaly_file, "ab") as log_file:
+        for i in range(len(list_of_protos)): 
+            # calculate time based on linear interpolation
+            proto_time = start_time + (duration - 0)/len(list_of_protos) * i   
+            proto = list_of_protos[i]
+
+            if random.random() > frequeny: 
```

Comment:
fix the seed of random module so it's easier to repeat these tests and debug them